src/game/Entities/Player.cpp

```diff
@@ -2040,6 +2040,8 @@ bool Player::ResolvePendingMount()
     }
 
     UpdateSpeed(MOVE_RUN, true); // update speed
+    if (m_pendingMountAuraFlying)
+        UpdateSpeed(MOVE_FLIGHT, true);
```

Comment:
It worked before because before it was done in place in aura. Now we are in an async environment and need to do it when client confirms the collision height check. Has to be done on unmount too however